server/src/test/java/org/elasticsearch/common/bytes/CompositeBytesReferenceTests.java

```diff
@@ -160,7 +160,53 @@ public void testGetIntLE() {
         assertThat(comp.getIntLE(2), equalTo(0x02010012));
         assertThat(comp.getIntLE(3), equalTo(0x03020100));
         assertThat(comp.getIntLE(4), equalTo(0x04030201));
-        Exception e = expectThrows(ArrayIndexOutOfBoundsException.class, () -> comp.getIntLE(5));
-        assertThat(e.getMessage(), equalTo("Index 4 out of bounds for length 4"));
+        // The jvm can optimize throwing ArrayIndexOutOfBoundsException by reusing the same exception,
+        // but these reused exceptions have no message or stack trace. This sometimes happens when running this test case.
+        // We can assert the exception message if -XX:-OmitStackTraceInFastThrow is set in gradle test task.
+        expectThrows(ArrayIndexOutOfBoundsException.class, () -> comp.getIntLE(5));
+    }
+
+    public void testGetDoubleLE() {
+        // first double = 1.2, second double = 1.4, third double = 1.6
+        byte[] data = new byte[] {
+            0x33,
+            0x33,
+            0x33,
+            0x33,
+            0x33,
+            0x33,
+            -0xD,
+            0x3F,
+            0x66,
+            0x66,
+            0x66,
+            0x66,
+            0x66,
+            0x66,
+            -0xA,
+            0x3F,
+            -0x66,
+            -0x67,
+            -0x67,
+            -0x67,
+            -0x67,
+            -0x67,
+            -0x7,
+            0x3F };
```

Comment:
This seems like a good use case for the `// tag::noformat` - `// end::noformat` syntax to disable automatic formatting for a section.  See the note in [CONTRIBUTING.md](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#java-language-formatting-guidelines).  Personally, I would put the bytes for each double on one line.